lib/src/man_up_service.dart

```diff
@@ -18,7 +18,7 @@ class ManUpService {
   ManUpService(
     this.url, {
     this.packageInfoProvider = const DefaultPackageInfoProvider(),
-    this.os,
+    required this.os,
```

Comment:
I would have preferred this to be set in the constructor lines, where if it was not passed it would fall back to `Platform.operatingSystem`

i.e:  ```
ManUpService(
    this.url, {
    this.packageInfoProvider = const DefaultPackageInfoProvider(),
    this.os,
    String? os,
    required http.Client http,
    ConfigStorage storage = const ConfigStorage(),
  })  : _client = HTTP, this.os = os ?? Platform.operatingSystem,
```

This would give the flexibility for tests on a specific platform, as well as making the current constructor a lot simpler to use. This would also allow the `os` string to be `final` as it should not be changing during runtime

Reply:
done.